packages/xo-server/src/api/xostor.mjs

```diff
@@ -105,10 +106,46 @@ formatDisks.resolve = {
   host: ['host', 'host', 'administrate'],
 }
 
-export async function create({ description, disksByHost, force, ignoreFileSystems, name, provisioning, replication }) {
+export const create = defer(async function (
+  $defer,
+  { description, disksByHost, force, ignoreFileSystems, name, provisioning, replication }
+) {
   const hostIds = Object.keys(disksByHost)
-  const hosts = hostIds.map(hostId => this.getObject(hostId, 'host'))
 
+  const tmpBoundObjectId = `tmp_${hostIds.join(',')}_${Math.random().toString(32).slice(2)}`
+
+  const xostorLicenses = await this.getLicenses({ productType: 'xostor' })
+
+  const now = Date.now()
+  const availableLicenses = xostorLicenses.filter(
+    ({ boundObjectId, expires }) => boundObjectId === undefined && (expires === undefined || expires > now)
+  )
+
+  let license = availableLicenses.find(license => license.productId === 'xostor')
```

Comment:
Why is it necessary to check the `productId`, isn't already handled by `getLicenses({ productId })`?

Reply:
My bad, I got them mixed up indeed!